### PR TITLE
Disabling the linebreak-style rule in .eslintrc.js

### PR DESCRIPTION
**Pull request by Atanda1**
Fixes creativecommons#649

I added "linebreak-style": 0, "global-require": 0, "eslint linebreak-style": [0, "error", "windows"] to the rule within the .eslintrc.js file to disable the linebreak-style rule, so npm run dev would work directly on windows.

---

Developer Certificate of Origin Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors. 1 Letterman Drive Suite D4700 San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this license document, but changing it is not allowed.

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I have the right to submit it under the open source license indicated in the file; or

(b) The contribution is based upon previous work that, to the best of my knowledge, is covered under an appropriate open source license and I have the right under that license to submit that work with modifications, whether created in whole or in part by me, under the same open source license (unless I am permitted to submit under a different license), as indicated in the file; or

(c) The contribution was provided directly to me by some other person who certified (a), (b) or (c) and I have not modified it.

(d) I understand and agree that this project and the contribution are public and that a record of the contribution (including all personal information I submit with it, including my sign-off) is maintained indefinitely and may be redistributed consistent with this project or the open source license(s) involved.

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -49,6 +50,10 @@ module.exports = {
     // allow debugger during development
     'no-debugger': process.env.NODE_ENV === 'production' ? 'error' : 'off',
     'brace-style': ["error", "stroustrup", { "allowSingleLine": true }],
+    // disable linebreak-style rule
+    'linebreak-style': 0,
+    "global-require": 0,
```
Comment (akmadian) [inline .eslintrc.js]: Please remove the global-require rule, it is not within the scope of a fix for the line break styling.

File: .eslintrc.js [hunk 2]
```diff
@@ -49,6 +50,10 @@ module.exports = {
...
+    "eslint linebreak-style": [0, "error", "windows"]
```
Comment (akmadian) [inline .eslintrc.js]: Please remove this line. `eslint linebreak-style` is not the name of any eslint rule so this effectively does nothing.

File: .eslintrc.js [hunk 3]
```diff
@@ -49,6 +50,10 @@ module.exports = {
     // allow debugger during development
     'no-debugger': process.env.NODE_ENV === 'production' ? 'error' : 'off',
     'brace-style': ["error", "stroustrup", { "allowSingleLine": true }],
+    // disable linebreak-style rule
+    'linebreak-style': 0,
```
Comment (akmadian) [inline .eslintrc.js]: Please change "0" to 'off', as the syntax is clearer and more in line with eslint syntax convention.

File: package.json [hunk 4]
```diff
@@ -116,6 +116,7 @@
     "node": ">= 6.0.0",
     "npm": ">= 3.0.0"
   },
+
```
Comment (akmadian) [inline package.json]: Please remove this whitespace

Comment (Atanda1) [reply package.json]: whitespace removed

Comment (akmadian) [reply package.json]: Apologies, I know I asked you to remove this whitespace, and I'm not sure what I was thinking when I asked. Please add the whitespace after the colon back :/

File: package.json [hunk 5]
```diff
@@ -113,9 +113,10 @@
     "webpack-node-externals": "^1.7.2"
   },
   "engines": {
-    "node": ">= 6.0.0",
-    "npm": ">= 3.0.0"
+    "node":">= 6.0.0",
```
Comment (akmadian) [inline package.json]: Please undo the removal of whitespace on lines 116 and 117

**Comment by Atanda1** [comment]
I've added the white space :)

**Comment by Atanda1** [comment]
Thanks! Would jump on the image height issue ASAP.

File: .eslintrc.js [hunk 6]
```diff
@@ -49,6 +50,8 @@ module.exports = {
     // allow debugger during development
     'no-debugger': process.env.NODE_ENV === 'production' ? 'error' : 'off',
     'brace-style': ["error", "stroustrup", { "allowSingleLine": true }],
+    // disable linebreak-style rule
+    'linebreak-style': 'off'
```
Comment (brenoferreira) [inline .eslintrc.js]: I'm not sure we should disable this. Did you try https://eslint.org/docs/rules/linebreak-style#using-this-rule-with-version-control-systems ?

Comment (Atanda1) [reply .eslintrc.js]: @akmadian

Comment (Atanda1) [reply .eslintrc.js]: Well, disabling it allows devs using windows OS to contribute directly on the project without having to run anything in the command line, or adding a folder. It'll be particularly useful to newbies who want to contribute without having to be confused just after running `npm run serve`, Instead we'll all be able to focus on the actual code we should be contributing. Or I could just open a new issue to adjust readme of this repo directing windows users on what to actually do after cloning the repo.

Comment (akmadian) [reply .eslintrc.js]: Yep, I'm on Windows and I have to add the rule to disable this every time I'm on a new branch, or wait for all the errors to get printed before the page actually gets served up.

Seems that there's no cross platform linebreak style in eslint, at least not that I could find. I don't think it's a good idea to mix styles, but if we choose one or the other, devs on either unix or windows systems will have to adapt. I'm not really sure what we should do about this. Thoughts @brenoferreira?

Comment (brenoferreira) [reply .eslintrc.js]: I think it's better to disable linting with the `webpack-dev-server` and have that integrated only on text editors. That way you won't have issues on the terminal when you run `npm run dev`.

We run that during CI build, so it shouldn't be a problem. Plus people should run it before pushing code to Github.

Comment (brenoferreira) [reply .eslintrc.js]: We should also set the appropriate settings on .gitattributes on the root dir
https://eslint.org/docs/rules/linebreak-style#using-this-rule-with-version-control-systems

**Comment by brenoferreira** [comment]
We've make some updates recently to the webpack and `npm run dev` doesn't show lint errors anymore. I think this is no longer relevant.

**Pull request closed without merging**